splunk_add_on_ucc_framework/ucc_ui_lib/src/main/webapp/components/TextComponent.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Text from '@splunk/react-ui/Text';
+
+class TextComponent extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    handleChange = (e, {value}) => {
+        this.props.handleChange(this.props.id, value);
+    };
+
+    render() {
+        return (
+            <Text
+                inline
+                placeholder={this.props.controlOptions.placeholder}
+                className={this.props.field}
+                disabled={this.props.disabled}
+                value={this.props.value}
+                onChange={this.handleChange}
+                type={this.props.encrypted === true ? 'password' : 'text'}
```

Comment:
@dkhatri-crest No need to check **this.props.encrypted === true**
We can use **this.props.encrypted ? 'password' : 'text'**

Reply:
Updated